### PR TITLE
Implement very basic secret sharing procedure

### PR DESCRIPTION
**Pull request by tudor-malene**
this is the mechanics of the secret-sharing protocol

### CODE REVIEW

File: ideas.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+1. How do you prevent multiple "genesis" nodes
+ - publish the hash of the shared secret  to the L1, and also added to the haeder of the rollup?
+ - hardcode the address of the management smart contract and each enclave can decide what to do when it ingests the blocks
+
+2. How to make the secret sharing more efficient?
```
Comment (otherview) [inline ideas.txt:5]: If the end goal is for the MC to handle the secret sharing do care about the effiency ? Reckon we can chat about this offline huh ?

Comment (tudor-malene) [reply ideas.txt:5]: you wouldn't want multiple aggregators to pay an L1 transaction for no good reason.

File: go/obscuronode/enclave/enclave.go [hunk 2]
```diff
@@ -308,6 +327,45 @@ func (e *enclaveImpl) Stop() {
 	e.exitCh <- true
 }
 
+func (e enclaveImpl) Attestation() common3.AttestationReport {
+	// Todo
+	return common3.AttestationReport{Owner: e.node}
+}
+
+// GenerateSecret - the genesis enclave is responsible with generating the secret entropy
+func (e enclaveImpl) GenerateSecret() common3.EncryptedSharedEnclaveSecret {
```
[the remark this answers is not in the document]
Comment (tudor-malene) [reply go/obscuronode/enclave/enclave.go:336]: good spot

File: go/obscuronode/enclave/enclave.go [hunk 3]
```diff
@@ -308,6 +327,45 @@ func (e *enclaveImpl) Stop() {
...
+	secret := make([]byte, 32)
+	n, err := rand.Read(secret)
+	if n != 32 || err != nil {
+		panic(fmt.Sprintf("Could not generate secret: %s", err))
```
[the remark this answers is not in the document]
Comment (tudor-malene) [reply go/obscuronode/enclave/enclave.go:340]: only once we're sure we know what to do with them. It's a bit early

**Pull request merged**